### PR TITLE
echo error messages to stderr

### PR DESCRIPTION
**Pull request by ryenus**
## Description

Title says it all, otherwise the error messages would be executed as code when the output
is supposed to be sourced, e.g. `$(flutter bash-completion)`

## Tests

I added the following tests:

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by ryenus** [comment]
@googlebot I signed it!

File: bin/flutter [hunk 1]
```diff
@@ -44,13 +44,13 @@ function retry_upgrade {
   local remaining_tries=$((total_tries - 1))
   while [[ "$remaining_tries" -gt 0 ]]; do
     (cd "$FLUTTER_TOOLS_DIR" && "$PUB" upgrade "$VERBOSITY") && break
-    echo "Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... ($remaining_tries tries left)"
+    >&2 echo "Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... ($remaining_tries tries left)"
```
Comment (gspencergoog) [inline bin/flutter]: I'd prefer the trailing version of this, as it seems more idiomatic. (here, and below)
```suggestion
    echo "Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... ($remaining_tries tries left)" >&2
```

**Review by gspencergoog — COMMENTED** [review]
Thanks for the PR!  Yes, you're right that these should all be sending to stderr instead.

File: bin/flutter (hunk 1 above)
Comment (ryenus) [reply bin/flutter]: done

**Review by christopherfujino — APPROVED** [review]
LGTM!

**Comment by Hixie** [comment]
This actually breaks a number of tests that expect this message to be on stdout (e.g. `dev/bots/analyze-sample-code.dart`, `dev/devicelab/bin/tasks/flutter_test_performance.dart`, `dev/devicelab/bin/tasks/run_release_test.dart`, `dev/devicelab/bin/tasks/dartdocs.dart`).

Also, this fix didn't have the corresponding change in the Windows batch file, so it still outputs to stdout (and having Windows and Linux be different will cause issues in the tests, that now have to handle both).

Also, this fix didn't have tests. We should make sure to test this so that once we've fixed it, we don't regress it (i.e. make sure that `$(flutter bash-completion)` works even when the batch file would output material).

I'll revert this fix for now. Please don't hesitate to resubmit the PR with the relevant tests changed.
It's possible that the right fix here is actually to make `flutter bash-completion` output to a file or a different file descriptor rather than using `stdout`?

**Comment by ryenus** [comment]
ehh, I think if there're tests broken due to this fix, it means those tests also need to be fixed, instead of this change being reverted. And it's actually a good thing to know which tests are broken.

**Comment by Hixie** [comment]
@ryenus we prefer to revert than leave things broken while waiting for fixes, but you are very welcome to submit a new PR that fixes the tests as well (and adds a test to make sure we don't revert the bug).

**Pull request merged**